smbc/context.c

```diff
@@ -551,28 +539,31 @@ Context_getxattr (Context *self, PyObject *args)
    *             https://technet.microsoft.com/en-us/library/cc961986.aspx
    */
 
-  size_t size = 375315;
-  buffer = (char *)malloc (size);
-  if(!buffer)
-    return PyErr_NoMemory ();
+  int ret;
+  char *uri = NULL;
+  char *name = NULL;
+  static smbc_getxattr_fn fn;
 
-  bzero(buffer, size);
+  // smbc_getxattr takes two string parameters
+  if (!PyArg_ParseTuple (args, "ss", &uri, &name))
+    {
+      return NULL;
+    }
+
+  char value[375315];
```

Comment:
Most linux environment set max stacksize to 8192k by default, so it seems no problem, but there may be issue with some embedded environments. @twaugh Which do you prefer stack or heap?